### PR TITLE
CLOUDSTACK-10109: Fix regression from PR #2295

### PR DESCRIPTION
**Pull request by rhtyd**
It seems a regression has come in, this PR/commit is to test what
fails/takes time on travis.

This is a stabilization PR to fix tests, included in 4.11.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@rhtyd isn't the smoke test PR already for this? can we close that one now?

**Comment by rhtyd** [comment]
@DaanHoogland this is to debug and fix the failures in Travis. I created this dummy PR to detect why job 3 has been failing since last few commits, all due to timeout issues.

**Comment by DaanHoogland** [comment]
Is see that @rhtyd . But i wonder why the smoke health test PR #2376 wasn't useable for that

**Comment by rhtyd** [comment]
@DaanHoogland using Travis/simulator I found regression introduced in pr #2295. Fixing the title and kick a test round.
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
Travis has passed finally! /cc @DaanHoogland @mike-tutowski

File: server/src/com/cloud/network/IpAddressManagerImpl.java [hunk 1]
```diff
@@ -964,13 +964,7 @@ public PublicIp doInTransaction(TransactionStatus status) throws InsufficientAdd
...
                         displayIp = vpc.isDisplay();
                     }
-                    PublicIp ip = fetchNewPublicIp(dcId, null, null, owner, VlanType.VirtualNetwork, guestNtwkId, isSourceNat, false, null, false, vpcId, displayIp, false);
-                    IPAddressVO publicIp = ip.ip();
-
-                    markPublicIpAsAllocated(publicIp);
-                    _ipAddressDao.update(publicIp.getId(), publicIp);
-
-                    return ip;
+                    return fetchNewPublicIp(dcId, null, null, owner, VlanType.VirtualNetwork, guestNtwkId, isSourceNat, true, null, false, vpcId, displayIp, false);
```
Comment (DaanHoogland) [inline server/src/com/cloud/network/IpAddressManagerImpl.java]: this is a revert of the code in #2295. Is that functionality now broken?

Comment (rhtyd) [reply server/src/com/cloud/network/IpAddressManagerImpl.java]: Yes, that PR incorrectly addressed merge conflict. One of the booleans changed from true to false, this fixes that.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

File: scripts/vm/hypervisor/kvm/setup_agent.sh [hunk 2]
```diff
@@ -224,17 +224,5 @@ then
...
 fi
 
-which aria2c
-if [ $? -gt 0 ]
-then
-    yum install epel-release -y
-    yum install aria2 -y
-    if [ $? -gt 0 ]
-    then
-        printf "failed to install aria2"
```
Comment (rhtyd) [inline scripts/vm/hypervisor/kvm/setup_agent.sh]: fyi - this is removed as installing cloudstack-agent will install aria2 and this script assumes a yum-enabled system that will fail on debian/ubuntu etc.

**Comment by DaanHoogland** [comment]
good work @rhtyd thanks for clearing my faulty uderstanding

**Comment by rhtyd** [comment]
Thanks @DaanHoogland are you lgtm on it? (pending test results of course)

**Comment by DaanHoogland** [comment]
lgtm

**Comment by rhtyd** [comment]
Merging this based on test results and code reviews.

**Pull request merged**